src/main/kotlin/gg/essential/essential-loader.gradle.kts

```diff
@@ -0,0 +1,64 @@
+package gg.essential
+
+import gg.essential.gradle.multiversion.Platform
+import gg.essential.gradle.util.RelocationTransform.Companion.registerRelocationAttribute
+
+
+plugins {
+    id("gg.essential.loom")
+}
+
+val platform = Platform.of(project)
+
+val essentialLoader: Configuration by configurations.creating
+
+when {
+    platform.isLegacyForge -> {
+        dependencies {
+            "runtimeOnly"(essentialLoader("gg.essential:loader-launchwrapper:1.1.3")!!)
+        }
+        tasks.named<Jar>("jar") {
+            dependsOn(essentialLoader)
+            from({ zipTree(essentialLoader.singleFile) })
+        }
+    }
+
+    platform.isFabric -> {
+        dependencies {
+            "include"("modRuntimeOnly"("gg.essential:loader-fabric:1.0.0")!!)
+        }
+    }
+
+    platform.isModLauncher -> {
+        dependencies {
+            val relocatedPackage = findProperty("essential.loader.package")?.toString() ?: throw GradleException("""
+                No essential loader package set.
+                You need to set `essential.loader.package` in the project's `gradle.properties` file to a specific package.
+                The recommended package is your mod's package.
+            """.trimIndent())
+            val relocationAttribute =
+                registerRelocationAttribute("essential-loader-relocated") {
+                    relocate("gg.essential.loader.stage0", "$relocatedPackage.stage0")
+                    // preserve stage1 path
+                    rename("gg/essential/loader/stage0/stage1.jar", "gg/essential/loader/stage0/stage1.jar")
+                }
+            essentialLoader.attributes {
+                attribute(relocationAttribute, true)
+            }
```

Comment:
There's a lot of places where this file is missing formatting, makes the closures look very messy. (New line after ```essentialLoader.attributes { ... }```)